waku.md

```diff
@@ -209,17 +209,29 @@ It is only relevant if you want to decrypt the incoming message, but if you only
 
 Data field contains encrypted message of the Envelope. In case of symmetric encryption, it also contains appended Salt (a.k.a. AES Nonce, 12 bytes). Plaintext (unencrypted) payload consists of the following concatenated fields: flags, auxiliary field, payload, padding and signature (in this sequence).
 
-	flags: 1 byte; first two bits contain the size of auxiliary field, third bit indicates whether the signature is present.
-	
-	auxiliary field: up to 4 bytes; contains the size of payload.
+Using [Augmented Backus-Naur form (ABNF)](https://tools.ietf.org/html/rfc5234) we have the following format:
+
+```
+bytes           = [repeat] OCTET
 
-	payload: byte array of arbitrary size (may be zero).
+; 1 byte; first two bits contain the size of auxiliary field, third bit indicates whether the signature is present.
+flags           = OCTET
 
-	padding: byte array of arbitrary size (may be zero).
+; contains the size of payload.
+auxiliary-field = bytes
 
-	signature: 65 bytes, if present.
+; byte array of arbitrary size (may be zero)
+payload         = bytes
 
-	salt: 12 bytes, if present (in case of symmetric encryption).
+; byte array of arbitrary size (may be zero).
+padding         = bytes
+
+; 65 bytes, if present.
+signature       = bytes
```

Comment:
This doesn't constraint the number of bytes... please see ABNF repeat `*` syntax

Reply:
same below